### DISCUSSION TITLE
Copilot Chat: Response redacted for no good reason rendering chat unusable in XML/XAML generation

### DISCUSSION THREAD

**Discussion by neurozero** [community/community#65222] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Note: Unlike mentioned in another (closed) discussion, this is not a minor issue. It happens often with XAML (and XML in general) and renders copilot unusable. (https://github.com/orgs/community/discussions/64160)

Edit: Adding the full message text for search: "The response was redacted to meet Responsible AI policies. If this was unexpected, please give feedback."

I'm not asking for anything illegal, immoral or weird. I'm asking for how to make a Grid have a two rows, one with a Toolbar and one with a GraphicsView. The resulting XML which I captured by recording my screen looks perfectly normal.
![copilot](https://github.com/community/community/assets/819839/a50f1a1d-e948-4fa0-806f-d1a09c4e9e79)
![copilot2](https://github.com/community/community/assets/819839/176b4a48-c051-4a48-b4f5-8460b6d536e6)

Please look into this, it has been happening for at least a couple of weeks.

**Comment by Akash1134** [comment]
Hi @neurozero ,

Thanks for reporting and apologies for the inconvenience. 

We have raised this with our engineering team internally.

**Comment by Kaddtechnologies** [comment]
What is the update on this? I have been experiencing this for a few days now and it is May 2024

**Comment by andreimanu** [comment]
Been experiencing it today, august 2024, so no luck

**Comment by ozrentk** [comment]
Same thing here. My question was "Show me an example of Menu added to MainWindow", for WPF. It started to generate XAML and failed with the same text. I tried with different prompts; it just fails to generate the XAML in the middle of the process.

I tried with prompt "are you able to generate any xaml without reducting to meet Responsible AI policies?"
Then it worked. Huh?

**Comment by jcookefcbt** [comment]
Have the same problem - I asked this question:
this code is using the standard Microsoft Logger, and not Serilog. I want it to be Serilog. Fix it
It started generating a response and then gave me thie same error. It is September 2024.

**Comment by mark74univpm** [comment]
I'm getting this message today with a simple and innocent question like this _"I need a C# function to create a random string from letters, digits and symbols"_

**Comment by simchor** [comment]
I have the same problem when asking to write error messages for end user in a json file format

**Comment by pstatxde** [comment]
I'm asking CoPilot to simply proofread the text on an "About Us" razor webpage and it's giving me this same error. Why?

**Comment by Drokkzz** [comment]
I get same error when i ask to create me a simple HTML/CSS template

**Comment by davaus** [comment]
I just got even after i asked it to rephrase so that it didn't get redacted....

**Comment by SimonGreener-at-Sylatech** [comment]
I get this problem with the prompt /doc, to generate a function header comment with a description of the function, for each parameter and the return value.  It is a valuable feature of the paid copilot subscription so it not working 25% of the time is really bad news.

**Comment by KevinDuganJr** [comment]
All I asked was if this stored procedure could be optimized and I get that redacted message. Oddly enough, when I was using the Free version, this didn't happen. I paid for a year and now it's basically without value to me. 🤷

**Reply by SimonGreener-at-Sylatech** [reply]
I had this type of problem *a lot*,  I resolved my issue by accident by closing all of the the Copilot threads I had started in VisualStudio (which was many).  Since then I have not had the message again.

**Comment by KotM** [comment]
The same issue for simply request "**Create dropdown list or popup when clicking this button.**"

Important note: it occurs when active document is XAML, but **works fine when switching to .cs**. Another observation: I'm using Copilot just for 2 weeks, but never encountered with such issue until I start working on implementing language switcher in my application. Not sure, but it seems to be related.

Just as additional information, selected part of code from my XAML:

```
        <ToolbarItem
            Clicked="OnLanguageSwitchClicked"
            Order="Primary"
            Priority="0"
            Text="{Static strings:CommonStrings.Language}">
            <ToolbarItem.IconImageSource>
                <FontImageSource
                    FontFamily="MaterialIconsOutlined-Regular"
                    Glyph="language"
                    Color="Black" />
            </ToolbarItem.IconImageSource>
        </ToolbarItem>
```

**Reply by Zhaph** [reply]
A couple of us on our enterprise plan have been having this issue intermittently as well. Switching from the default model to GPT-4o seemed to work around it.
